### ISSUE TITLE
Upgrade to Spring Boot 1.4.1 release

### ISSUE THREAD

**Issue by jclagache** [jclagache/spring-data-mybatis#9]
Upgrade spring-data-mybatis to be [Spring Boot](https://projects.spring.io/spring-boot/) 1.4.1 compliant.

**Comment by kopax** [comment]
I don't have previous experiences with spring-data but I have documented myself in order to build this module. I have found a few interesting implementation related to the mybatis implementation of spring-data.  Unfortunately, none of these run with the latest spring version

For example, this repo has PaginatedRepository and CrudRepository:

https://github.com/boyley/Application-erp

I am a loyal user of mybatis and after reading the spring-data tutorial, I realized how important this module is for spring + mybatis user community.

Anyhow, I need to release an API pretty soon, should I swap to JPA or not, I don't wan't to so this is why I am available to work with you. I have already checkout out the next release branch to see how I can help.
I don't wan't to use Github as a forum so it is the last time I say it. Thanks for creating the branch btw!
